HyperFIDO Pro Mini ctap2token: CBOR error: CTAP2_ERR_INVALID_CBOR

I'm trying a [HyperFIDO Pro Mini authenticator](https://www.hypersecu.com/hyperfido) ([amazon buy link](https://www.amazon.es/gp/product/B0813YWZB2)) with the https://github.com/flynn/u2f/tree/webauthn branch that is being worked on by @daeMOn63 at https://github.com/flynn/u2f/pull/12 and its failed with a `CBOR error`.

I've tried the [ctap2token example](https://github.com/flynn/u2f/blob/webauthn/ctap2token/example/main.go) as:

```bash
git clone -b webauthn https://github.com/flynn/u2f.git
cd u2f
go run ctap2token/example/main.go
```

Which returned:

```
Token info:
&ctap2token.GetInfoResponse{Versions:[]string{"U2F_V2", "FIDO_2_0", "FIDO_2_1_PRE"}, Extensions:[]string{"credProtect", "hmac-secret"}, AAGUID:[]uint8{0x9f, 0x77, 0xe2, 0x79, 0xa6, 0xe2, 0x4d, 0x58, 0xb7, 0x0, 0x31, 0xe5, 0x94, 0x3c, 0x6a, 0x98}, Options:ctap2token.AuthenticatorOptions{"clientPin":false, "credentialMgmtPreview":true, "plat":false, "rk":true, "up":true}, MaxMsgSize:0x800, PinProtocol:[]uint{0x1}}
Sending makeCredential request, please press authenticator button...
Successfully created credential
MakeCredentials signature is valid!
credentialID: d8ac1cf85bbecfb6a3affebdd0...
Sending GetAssertion request, please press authenticator button...
panic: ctap2token: CBOR error: CTAP2_ERR_INVALID_CBOR

goroutine 1 [running]:
main.main()
	/home/rgl/webauthn/u2f/ctap2token/example/main.go:137 +0x1433
exit status 2
```

I pressed the button once and it go thru the first step successfully; but when it reached the second step of asking me to press the button, it failed with the `CBOR error` before I pressed the button a second time.

Can you help me troubleshoot this? Is there something else I should try?

--- a/ctap2token/example/main.go
+++ b/ctap2token/example/main.go
@@ -123,9 +123,8 @@ func main() {
 			RPID: "example.com",
 			AllowList: []*ctap2token.CredentialDescriptor{
 				{
-					ID:         mcpAuthData.AttestedCredentialData.CredentialID,
-					Transports: []ctap2token.AuthenticatorTransport{ctap2token.USB},
-					Type:       ctap2token.PublicKey,
+					ID:   mcpAuthData.AttestedCredentialData.CredentialID,
+					Type: ctap2token.PublicKey,
 				},
 			},
 			ClientDataHash: clientDataHash,

--- a/ctap2token/token.go
+++ b/ctap2token/token.go
@@ -629,9 +629,10 @@ const (
 // CredentialDescriptor defines a credential returned by the authenticator,
 // as defined by https://www.w3.org/TR/webauthn/#credential-dictionary
 type CredentialDescriptor struct {
-	ID         []byte                   `cbor:"id"`
-	Type       CredentialType           `cbor:"type"`
-	Transports []AuthenticatorTransport `cbor:"transports"`
+	ID   []byte         `cbor:"id"`
+	Type CredentialType `cbor:"type"`
+	// Don't set transports field when using HyperSecu Mini tokens.
+	Transports []AuthenticatorTransport `cbor:"transports,omitempty"`
 }
 
 // AuthenticatorTransport defines hints as to how clients might communicate with a particular authenticator,

